kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/InBandKeyManagerTest.java

```diff
@@ -270,6 +318,47 @@ void afterWeFailToLoadADekTheNextEncryptionAttemptCanSucceed() {
         assertThat(encrypted).hasSize(2);
     }
 
+    @Test
+    void afterWeFailToDecryptAnEDekTheNextEncryptionAttemptCanSucceed() {
```

Comment:
From my commit:

> Note: the testing required is ugly, Sam commented on this for the EDEK
> PR too. It's tough because the InBandKeyManager knows the parcel format
> and we don't have test utilities to independently generate examples of
> parcels yet. If we had the ability to generate serialized record examples
> we could avoid going through an encrypt cycle with a real KMS implementation
> just to get properly serialized records.

But keen to keep it separate